src/main/java/spoon/template/TemplateMatcher.java

```diff
@@ -185,7 +186,20 @@ public void scan(CtElement element) {
 				}
 				super.scan(element);
 			}
-		}.scan(targetRoot);
+		};
+
+		scanner.scan(templateRoot);
+		if (!finds.contains(templateRoot)) {
+			throw new SpoonException("TemplateMatcher was unable to find itself, it certainly express a problem somewhere in the template.");
```

Comment:
TemplateMatcher was unable to find itself, it certainly indicates a bug. Please revise your template or report an issue.